pkg/runner/local_docker.go

```diff
@@ -787,80 +452,6 @@ func newDataNetwork(ctx context.Context, cli *client.Client, rw *rpc.OutputWrite
 	return id, subnet, err
 }
 
-// ensure container is started
-func ensureInfraContainer(ctx context.Context, cli *client.Client, ow *rpc.OutputWriter, containerName string, imageName string, networkID string, pull bool, cmds ...string) (id string, err error) {
-	container, _, err := docker.EnsureContainer(ctx, ow, cli, &docker.EnsureContainerOpts{
-		ContainerName: containerName,
-		ContainerConfig: &container.Config{
-			Image: imageName,
-			Cmd:   cmds,
-		},
-		HostConfig: &container.HostConfig{
-			NetworkMode:     container.NetworkMode(networkID),
-			PublishAllPorts: true,
-			Resources: container.Resources{
-				Ulimits: []*units.Ulimit{
-					{Name: "nofile", Hard: InfraMaxFilesUlimit, Soft: InfraMaxFilesUlimit},
-				},
-			},
-		},
-		PullImageIfMissing: pull,
-	})
-	if err != nil {
-		return "", err
-	}
-
-	return container.ID, err
-
-}
-
-// ensureSidecarContainer ensures there's a testground-sidecar container started.
-func ensureSidecarContainer(ctx context.Context, cli *client.Client, workDir string, ow *rpc.OutputWriter, controlNetworkID string) (id string, err error) {
-	dockerSock := "/var/run/docker.sock"
-	if host := cli.DaemonHost(); strings.HasPrefix(host, "unix://") {
-		dockerSock = host[len("unix://"):]
-	} else {
-		ow.Warnf("guessing docker socket as %s", dockerSock)
-	}
-	container, _, err := docker.EnsureContainer(ctx, ow, cli, &docker.EnsureContainerOpts{
-		ContainerName: "testground-sidecar",
-		ContainerConfig: &container.Config{
-			Image:      "ipfs/testground:latest",
-			Entrypoint: []string{"testground"},
-			Cmd:        []string{"sidecar", "--runner", "docker", "--pprof"},
-			Env:        []string{"REDIS_HOST=testground-redis", "GODEBUG=gctrace=1"},
-		},
-		HostConfig: &container.HostConfig{
-			PublishAllPorts: true,
-			PortBindings:    nat.PortMap{"6060": []nat.PortBinding{{HostIP: "0.0.0.0", HostPort: "0"}}},
-			NetworkMode:     container.NetworkMode(controlNetworkID),
-			// To lookup namespaces. Can't use SandboxKey for some reason.
-			PidMode: "host",
-			// We need _both_ to actually get a network namespace handle.
-			// We may be able to drop sys_admin if we drop netlink
-			// sockets that we're not using.
-			CapAdd: []string{"NET_ADMIN", "SYS_ADMIN"},
-			// needed to talk to docker.
-			Mounts: []mount.Mount{{
-				Type:   mount.TypeBind,
-				Source: dockerSock,
```

Comment:
You also hardcoded this to "/var/run/docker.sock". Not everybody runs Linux or the default docker configuration. Please be careful not to cause any regressions with your refactors, @coryschwartz!